### PR TITLE
Feature/profile

### PR DESCRIPTION
**Pull request by AnaraAlmazova**
Update  profile

### CODE REVIEW

File: src/main/java/com/peaksoft/giftlistm5/model/User.java [hunk 1]
```diff
@@ -109,4 +109,13 @@ public boolean isCredentialsNonExpired() {
...
         return true;
     }
+
+
+    public void setUserInfo(UserInfo userInfo) {
+    }
+
+
+    public UserInfo getUserInfo() {
+        return getUserInfo();
```
Comment (MamatisaevaNurisa) [inline src/main/java/com/peaksoft/giftlistm5/model/User.java]: I am not sure, but I think here return UserInfo not getUserInfo.

File: src/main/java/com/peaksoft/giftlistm5/model/UserInfo.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+    @OneToOne(cascade ={CascadeType.PERSIST,CascadeType.MERGE,CascadeType.DETACH,CascadeType.REFRESH},mappedBy = "userInfo")
+    private User user;
+
+    private String facebookLink;
+    private String instagramLink;
+    private String telegramLink;
+    private String vkLink;
+}
+
```
Comment (MamatisaevaNurisa) [inline src/main/java/com/peaksoft/giftlistm5/model/UserInfo.java]: Remove extra lines

File: src/main/java/com/peaksoft/giftlistm5/service/UserProfileService.java [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+        response.setPhoneNumber(user.getUserInfo().getPhoneNumber());
+        response.setShoeSize(user.getUserInfo().getShoeSize());
+        response.setDateOfBirth(user.getUserInfo().getDateOfBirth());
+        response.setFacebookLink(user.getUserInfo().getFacebookLink());
+        response.setInstagramLink(user.getUserInfo().getInstagramLink());
+        response.setTelegramLink(user.getUserInfo().getTelegramLink());
+        response.setVkLink(user.getUserInfo().getVkLink());
+        return response;
+    }
+
```
Comment (MamatisaevaNurisa) [inline src/main/java/com/peaksoft/giftlistm5/service/UserProfileService.java]: remove lines

Comment (MamatisaevaNurisa) [reply src/main/java/com/peaksoft/giftlistm5/service/UserProfileService.java]: method convertToResponse and getFullInfoMyProfile return the same value.

File: src/main/java/com/peaksoft/giftlistm5/service/UserProfileService.java [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+        if (request.getImage() == null){
+            user.setImage("https://fsd.multiurok.ru/html/2022/01/21/s_61ea847d0c3d7/php2JwfX3_Iskusstvo-darit-podarki_html_157b25f0349d7d3a.jpg");
+        }else {
+            user.setImage(request.getImage());
+        }
+        try {
+            user.setCountry(Country.valueOf(request.getCountry()));
+            user.setGender(Gender.valueOf(request.getGender().toUpperCase()));
+            Shoes shoes = Shoes.FORTY_ONE;
```
Comment (Khaadikg) [inline src/main/java/com/peaksoft/giftlistm5/service/UserProfileService.java:46]: Default value должно ли быть 41?

File: src/main/java/com/peaksoft/giftlistm5/service/UserProfileService.java [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+        return myProfileResponse;
+    }
+    public ProfileResponse getFullProfile() {
+        User user = getAuthPrincipal();
+        return mapToProfileResponse(user);
+    }
+    public ProfileResponse changePassword(String oldPassword, String newPassword, String checkNewPassword){
+        User user = getAuthPrincipal();
+        if (!newPassword.equals(checkNewPassword)) throw new IncorrectLoginException("Passwords must be same!");
+        if (user.getPassword().equals(encoder.encode(oldPassword))) {
```
Comment (Khaadikg) [inline src/main/java/com/peaksoft/giftlistm5/service/UserProfileService.java:81]: Нужна проверка на OldPassword = repostitoryGetOldPassword

**Review by Khaadikg — CHANGES_REQUESTED** [review]
Fine

**Pull request merged**